### PR TITLE
Change the Debian example to reflect standard Debian naming.

### LINKED ISSUE #143: example package name does not follow Debian naming conventions.
This line appears to be incorrect

https://github.com/bazelbuild/rules_pkg/blob/master/pkg/pkg.bzl

`line 375  out_deb = "%s_%s_%s.deb" % (package, version, architecture)
`
The the name is not following the correct syntax.
`<foo>_<VersionNumber>-<DebianRevisionNumber>_<DebianArchitecture>.deb 
`
(https://www.debian.org/doc/manuals/debian-faq/ch-pkg_basics.en.html)

foo_VVV-RRR_AAA.deb (where VVV, RRR and AAA are the version, revision and architecture of the package in question, respectively)
Package names (both source and binary, see Package) must consist only of lower case letters (a-z), digits (0-9), plus (+) and minus (-) signs, and periods (.). They must be at least two characters long and must start with an alphanumeric character.
Regular expressions are not currently supported. however some simple checks should be possible for example:

```
if '_' not in name:
   for x in name.elems():
     if x.isupper():
          fail("Uppercase characters are not allowed in package names", "name")
     elif x in "+-.":
            continue
     elif not x.alnum():
          fail("Only lowercase characters and numbers are allowed in package names", "name")
     else
        fail("_ is not allowed in package names", "name")
               

```

### PR DESCRIPTION
**Pull request by aiuto**
Closes #143

### CODE REVIEW

File: examples/naming_package_files/readme.md [hunk 1]
```diff
@@ -64,3 +64,41 @@ names_from_toolchains = rule(
 bazel build :example2
 ls -l bazel-bin/example2*.tar
 ```
+
```
Comment (nacl) [inline examples/naming_package_files/readme.md:67]: Consider renaming this file to `README.md` in a future change to match existing conventions.

**Review by nacl — APPROVED** [review]
Comments are all nitty markdown/naming suggestions, some of which can be deferred if you like.  LGTM otherwise.

File: examples/naming_package_files/readme.md [hunk 2]
```diff
@@ -64,3 +64,41 @@ names_from_toolchains = rule(
...
+### Debian package names
+
+Debian package names are of the form `<package>_<version>-<revision>_<arch>.deb`.
+
+One way you might do that is shown in this snipped from the `BUILD` file.
+
+```
```
[the remark this answers is not in the document]
Comment (aiuto) [reply examples/naming_package_files/readme.md:74]: Done

File: examples/naming_package_files/readme.md [hunk 3]
```diff
@@ -64,3 +64,41 @@ names_from_toolchains = rule(
...
+    package = "foo-tools",
+    ...
+    # Note: target_cpu comes from the --cpu on the command line, and dows not
+    # have to be stated in the BUILD file.
+    package_file_name = "foo-tools_{version}-{revision}_{target_cpu}.deb",
+    package_variables = ":my_naming_vars",
+    version = VERSION,
+)
+```
+Try building 'bazel build :a_deb_package` then examine the results. Note that
```
[the remark this answers is not in the document]
Comment (aiuto) [reply examples/naming_package_files/readme.md]: Oh. Good catch. That is clearly a typo.

File: examples/naming_package_files/readme.md (hunk 1 above)
Comment (aiuto) [reply examples/naming_package_files/readme.md:67]: ack

**Pull request merged**